scipy/sparse/tests/test_base.py

```diff
@@ -756,23 +756,39 @@ def test_sum(self):
         matrices = [dat_1, dat_2, dat_3, dat_4, dat_5]
 
         def check(dtype, j):
+            if j == 1:
+                # At the lowest tested precision for floats and complex
+                # numbers, the sparse matrix sum method can lead to different
+                # results in the lower decimals from the dense matrix sum
+                # method. This necessitaties a more lenient comparison between
+                # the resulting matrices.
+                if dtype == np.complex64:
+                    decimal = 4
+                else:
+                    decimal = 5
```

Comment:
@perimosocordiae How about using smaller random float matrices in `test_sum`, e.g. `np.random.rand(5, 5)` instead of `np.random.rand(40, 40)`? That sidesteps all precision issues in the sparse vs. dense sum comparisons, too.